File: dynamic_listings/C++/default/optimization/lazy_mode.cpp
class LazyObject{
    private:
        int* numbers_reference;

    public:
        LazyObject(int[] numbers){
            // Saves the original list (possibly as a reference)
            numbers_reference = numbers;
        }

        int getObject(int index){
            // Calculates the halved number on-demand
            return numbers_reference[index] / 2;
        }
};

File: dynamic_listings/rust/default/optimization/lazy_mode.rs
struct LazyObject {
    numbers_reference: Vec<u32>,
}

impl LazyObject {
    pub fn new(numbers: &[u32]) -> Self {
        // Saves the original list (possibly as a reference)
        Self {
            numbers_reference: numbers.to_vec(),
        }
    }

    pub fn object(&self, index: usize) -> u32 {
        // Calculates the halved number on-demand
        self.numbers_reference[index] / 2
    }
}
